### PR TITLE
Scan alerts once per second and transfer items to the logistic network

### PR DESCRIPTION
**Pull request by bengardner**
This only handles a single force ("player"), but that should work OK, since the item-network does not support multiple forces.

### CODE REVIEW

File: src/NetworkChest.lua [hunk 1]
```diff
@@ -609,6 +609,62 @@ function M.onTick()
...
+function M.handle_missing_material(entity, name)
+  -- did we already transfer something for this ghost/upgrade?
+  if GlobalState.alert_transfer_get(entity.unit_number) ~= true then
+    -- We can only do something about entities in a network
+    -- REVISIT: assuming "player" force only
+    local net = entity.surface.find_logistic_network_by_position(entity.position, "player")
+    if net ~= nil and net.available_construction_robots > 0 then
+      local network_count = GlobalState.get_item_count(name)
+      if network_count > 0 then
+        local n_inserted = net.insert({name=name, count=1})
```
Comment (year6b7a) [inline src/NetworkChest.lua:702]: Why only insert one item?

File: src/NetworkChest.lua [hunk 2]
```diff
@@ -609,6 +609,62 @@ function M.onTick()
...
+  M.updatePlayers()
+  M.check_alerts()
+end
+
+function M.handle_missing_material(entity, name)
+  -- did we already transfer something for this ghost/upgrade?
+  if GlobalState.alert_transfer_get(entity.unit_number) ~= true then
+    -- We can only do something about entities in a network
+    -- REVISIT: assuming "player" force only
+    local net = entity.surface.find_logistic_network_by_position(entity.position, "player")
```
Comment (year6b7a) [inline src/NetworkChest.lua:698]: What happens if there are multiple logistic networks in the surface and the player is standing in the logistic network that doesn't need items?

File: src/NetworkChest.lua (hunk 1 above)
Comment (bengardner) [reply src/NetworkChest.lua:702]: Each alert is for one missing item, which points to one entity.

I track "handled" alerts by recording the entity.unit_number. I don't want to send an item twice for the same ghost.

I could combine all outstanding alerts and request a bunch at the same time, but that gets complicated.
How would I track which alerts have been handled?

The code has to handle the case where I add an item to the network, but it gets used for a different purpose.
That's what the timeout is for.

File: src/NetworkChest.lua (hunk 2 above)
Comment (bengardner) [reply src/NetworkChest.lua:698]: You are talking about a entity that has coverage from multiple "construction" logistic networks zone for cells that do not connect via "logistic" robots.

I assume it wouldn't show an alert in that case, as the request would be satisfied by the network that has the item.
The logistic network is based on the ghost position, not the player

If there are multiple networks that have coverage and none have the item, then it shouldn't matter which network gets the item.

I suppose there are corner cases where one network has construction robots and the other doesn't.
In that case, if `find_logistic_network_by_position()` return the network with no robots, then nothing will get added to that network and the alert will stay active.

File: src/NetworkChest.lua (hunk 1 above)
Comment (year6b7a) [reply src/NetworkChest.lua:702]: After a second read-through this makes sense. Those transfers should be cheap, but might need revisiting if performance somehow becomes an issue if players are constructing a ton of entities at once.

File: src/NetworkChest.lua (hunk 2 above)
Comment (year6b7a) [reply src/NetworkChest.lua:698]: Got it, thanks for explaining. I tried out this branch and it makes a lot more sense.

**Pull request merged**